lms/djangoapps/course_blocks/transformers/hidden_content.py

```diff
@@ -0,0 +1,99 @@
+"""
+Visibility Transformer implementation.
+"""
+from openedx.core.lib.block_structure.transformer import BlockStructureTransformer, FilteringTransformerMixin
+from xmodule.seq_module import SequenceModule
+from .utils import collect_merged_boolean_field, collect_merged_date_field
+
+
+class HiddenContentTransformer(FilteringTransformerMixin, BlockStructureTransformer):
+    """
+    A transformer that enforces the hide_after_due field on
+    blocks by removing children blocks from the block structure for
```

Comment:
FYI - I updated the transformer code to no longer assume that the `hide_after_due` field is set on subsections.  In doing so, this means the subsection itself will also no longer be available in the course blocks API.  This should be fine for now and is consistent with what we are doing in the `ProctoredExamTransformer`, which @sanfordstudent is generalizing.